### DISCUSSION TITLE
accept files in Flux CLI's 'push artifact' subcommand

### DISCUSSION THREAD

**Discussion by developer-guy** [fluxcd/flux2#3176] [Proposals] [answered]
Flux CLI takes a `--path` parameter that can be used to specify a directory where the Kubernetes manifests are located to push manifests to an OCI registry. Still, some projects keep all the manifests within the same file; for example, Kyverno keeps all the deployment resources on a single file, [install.yaml](https://github.com/kyverno/kyverno/blob/main/config/install.yaml). In that case, to push this manifest, I need to create a directory, copy this file to that directory and tell Flux to look at that directory. To avoid that, maybe Flux CLI can also accept a flag called `--file` to look for a file instead of a directory.

**Comment by stefanprodan** [comment, marked as answer]
We could make `--path` accept a file path, copy the file to `/tmp/random-dir/` then package it as if we were running `flux push artifact --path ./` from inside the tmp dir.

**Reply by developer-guy** [reply]
Yeah, that makes a lot of sense to me. I'm willing to do this. May I do it?

**Reply by stefanprodan** [reply]
Sure, please create an issue for both `build` and `push` in flux2. First step is to add file support here https://github.com/fluxcd/pkg/tree/main/oci/client and tests. Thanks!
